### DISCUSSION TITLE
RFC-5914

### DISCUSSION THREAD

**Discussion by jsatchell** [librasn/rasn#115] [Ideas]
I am working on a small private project to learn some rust, and see how some of my day job code could be re-implemented. One of things that is used for a proprietary customer (their copyright) is Trust Anchors, as defined by RFC 5914 . This is a small ASN1 module, mostly re-using types defined elsewhere in ras-pkix. 

I have had a naive attempt at rolling my own definition for RFC-5914, copying the style of the definitions in rasn-pkix.

Would their be any interest or value in this for a wider audience? If so, I can ask my boss about releasing what I have done so far. There are less than 50 lines.

I am pretty new to this, so please forgive me if this is not suitable question for this forum!

**Comment by XAMPPRocky** [comment]
Thank you for your question! I'd accept a PR adding it to `rasn-pkix`, it seems too small to be its own crate. Contributions and questions are always appreciated and encouraged 🙂

**Reply by jsatchell** [reply]
There is a PR; not really much to it. I have seen at least one embedded case where the TA is not a full certificate, but rather a very cut down option allowed by this RFC. The RFC also defines an OID for a new content type, which I added. Bet that one is not met very often in the wild!

**Reply by XAMPPRocky** [reply]
>  I have seen at least one embedded case where the TA is not a full certificate, but rather a very cut down option allowed by this RFC.

Is that something that your PR can parse? I don't want to support non-standard stuff, but if it's allowed by the RFC then maybe it should be somewhat compatible.

**Comment by jsatchell** [comment]
It should handle that case. That would correspond to the TaInfo choice, and
many fields are optional, so very simple cases are supported.
